integrade/api.py

```diff
@@ -0,0 +1,216 @@
+    response body as JSON and return the result.
+    """
+    raise_error_for_status(response)
+    return response.json()
+
+
+class Client(object):
+    """A client for interacting with the cloudigrade API.
+
+    This class is a wrapper around the ``requests.api`` module provided by
+    `Requests`_. Each of the functions from that module are exposed as methods
+    here, and each of the arguments accepted by Requests' functions are also
+    accepted by these methods. The difference between this class and the
+    `Requests`_ functions lies in its configurable request and response
+    handling mechanisms.
+
+    All requests made via this client use the base URL of the Cloudigrade
+    server provided in your the environment variable $CLOUDIGRADE_BASE_URL.
+
+    Additionally, we need an authorizatoin token. At the moment this must be
+    provided to us via $CLOUDIGRADE_TOKEN. This is likely to change as the API
+    develops.
+
+    You can override this base url by assigning a new value to the url
+    field.
+
+    Example::
+        >>> from integrade import api
+        >>> client = api.Client()
+        >>> # I can now make requests to the Cloudigrade server
+        >>> # using relative paths, because the base url is
+        >>> # was set using information from my enviroment.
+        >>>
+        >>> client.get()
+        >>>
+        >>> # now if I want to do something else,
+        >>> # I can change the base url
+        >>> client.url = 'https://www.whatever.com'
+
+    .. _Requests: http://docs.python-requests.org/en/master/
+    """
+
+    def __init__(self, response_handler=None, url=None, authenticate=True):
+        """Initialize this object, collecting base URL from config file.
+
+        If no response handler is specified, use the `code_handler` which will
+        raise an exception for 'bad' return codes.
+
+        If no URL is specified, then the url will be built from the
+        environment variables $CLOUDIGRADE_BASE_URL and $USE_HTTPS values (see
+        integrade/config.py).
+        """
+        self.url = url
+        self.token = None
+        cfg = config.get_config()
+        self.verify = cfg.get('ssl-verify', False)
+
+        if not self.url:
+            hostname = cfg.get('base_url')
+
+            if not hostname:
+                raise exceptions.BaseUrlNotFound(
+                    'Make sure you have $CLOUDIGRADE_BASE_URL set in in'
+                    ' your environment.'
+                )
+
+            scheme = cfg.get('scheme')
+            self.url = urlunparse(
+                            (scheme,
+                             hostname,
+                             'api/v1/', '', '', '')
+                            )
+
+        if response_handler is None:
+            self.response_handler = code_handler
+        else:
+            self.response_handler = response_handler
+
+        if authenticate:
+            self.token = cfg.get('superuser_token')
```

Comment:
If there is no token on the config file and I ask for authentication it won't really authenticate. It would be better to provide some feedback for the one running the tests.